Tools/machines/frontier-olcf/submit.sh

```diff
@@ -6,12 +6,155 @@
+        if [ -f libamdhip64.so.5 ]; then ln -s libamdhip64.so.5 libamdhip64.so; fi"
+    # You may notice that some libraries are still linked
+    # from /sw/frontier, even after SBCASTing.
+    # This is because the Spack-build modules use RPATH to find their dependencies.
+    # This behavior cannot be changed.
+    echo "*****ldd /mnt/bb/$USER/warpx*****"
+    ldd /mnt/bb/$USER/warpx
+    echo "*************************************"
+
+else
+    # Find out the path of the pywarpx module
+    PYWARPX_PATH=$(python3 -c "import pywarpx; print(pywarpx.__path__[0])")
+
+    if [ ${warpx_dim} == "1" ]; then
+        export WARPX_SO="warpx_pybind_1d.cpython-39-x86_64-linux-gnu.so"
+    elif [ ${warpx_dim} == "2" ]; then
+        export WARPX_SO="warpx_pybind_2d.cpython-39-x86_64-linux-gnu.so"
+    elif [ ${warpx_dim} == "3" ]; then
+        export WARPX_SO="warpx_pybind_3d.cpython-39-x86_64-linux-gnu.so"
+    elif [ ${warpx_dim} == "RZ" ]; then
+        export WARPX_SO="warpx_pybind_rz.cpython-39-x86_64-linux-gnu.so"
+    else
+        echo "${warpx_dim} is not a valid WarpX dimensionality!"
+        exit 1
+    fi
+
+    srun -N ${SLURM_NNODES} -n ${SLURM_NNODES} --ntasks-per-node=1 \
+        bash -c "mkdir /mnt/bb/$USER/pywarpx/"
+
+    # Broadcast all the files of the pywarpx module to the nodes
+    for FILE in $PYWARPX_PATH/*
+    do
+        if [ ! -d "$FILE" ]; then
+                # Broadcasting the libraries linked by WarpX.
+            if [ $(basename $FILE) == $WARPX_SO ]; then
+                sbcast --send-libs --exclude=NONE -pf $FILE /mnt/bb/$USER/pywarpx/$(basename $FILE)
+                if [ ! "$?" == "0" ]; then
+                    echo "SBCAST failed!"
+                    exit 1
+                fi
+            else
+                sbcast -pf $FILE /mnt/bb/$USER/pywarpx/$(basename $FILE)
+                if [ ! "$?" == "0" ]; then
+                    echo "SBCAST failed!"
+                    exit 1
+                fi
+            fi
+        fi
+    done
+
+    # All required libraries should be here now
+    export LD_LIBRARY_PATH="/mnt/bb/$USER/pywarpx/${WARPX_SO}_libs"
+
+    # libfabric dlopen's several libraries:
+    export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:$(pkg-config --variable=libdir libfabric)"
+    # cray-mpich dlopen's libhsa-runtime64.so and
+    # libamdhip64.so (non-versioned), so symlink on each node:
+    srun -N ${SLURM_NNODES} -n ${SLURM_NNODES} --ntasks-per-node=1 --label -D /mnt/bb/$USER/pywarpx/${WARPX_SO}_libs \
+        bash -c "if [ -f libhsa-runtime64.so.1 ]; then ln -s libhsa-runtime64.so.1 libhsa-runtime64.so; fi;
+        if [ -f libamdhip64.so.5 ]; then ln -s libamdhip64.so.5 libamdhip64.so; fi"
+
+    # Add broadcasted python libraries to PYTHONPATH
+    export PYTHONPATH=/mnt/bb/$USER/:$PYTHONPATH
+
+    echo "*************************************"
+    python3 -c "import pywarpx; print('pywarpx.__path__: ', pywarpx.__path__)"
```

Comment:
Execute this *after* `ldd`?

Currently, if the import fails, we will not know the output of `ldd`, which might be helpful.